The class `sun.misc.Unsafe` is not avilable in JDK 11

The `Striped64` class depends on `sun.misc.Unsafe` which has been removed in JDK 11, resulting in a class not found error on loading when using Jigsaw modules.

```
Exception in thread "main" java.lang.NoClassDefFoundError: sun/misc/Unsafe
	at simpleclient@0.6.0/io.prometheus.client.Striped64.getUnsafe(Striped64.java:316)
	at simpleclient@0.6.0/io.prometheus.client.Striped64.<clinit>(Striped64.java:296)
	at simpleclient@0.6.0/io.prometheus.client.Gauge$Child.<init>(Gauge.java:137)
	at simpleclient@0.6.0/io.prometheus.client.Gauge.newChild(Gauge.java:98)
	at simpleclient@0.6.0/io.prometheus.client.Gauge.newChild(Gauge.java:66)
	at simpleclient@0.6.0/io.prometheus.client.SimpleCollector.labels(SimpleCollector.java:76)
	at simpleclient@0.6.0/io.prometheus.client.SimpleCollector.initializeNoLabelsChild(SimpleCollector.java:107)
	at simpleclient@0.6.0/io.prometheus.client.SimpleCollector.<init>(SimpleCollector.java:174)
	at simpleclient@0.6.0/io.prometheus.client.Gauge.<init>(Gauge.java:69)
	at simpleclient@0.6.0/io.prometheus.client.Gauge$Builder.create(Gauge.java:75)
	at simpleclient@0.6.0/io.prometheus.client.Gauge$Builder.create(Gauge.java:72)
	at simpleclient@0.6.0/io.prometheus.client.SimpleCollector$Builder.register(SimpleCollector.java:245)
	at simpleclient@0.6.0/io.prometheus.client.SimpleCollector$Builder.register(SimpleCollector.java:238)
	at ca.on.oicr.gsi.runscanner.server/ca.on.oicr.gsi.runscanner.server.util.Scheduler.<clinit>(Scheduler.java:160)
	at ca.on.oicr.gsi.runscanner.server/ca.on.oicr.gsi.runscanner.server.Server.main(Server.java:26)
Caused by: java.lang.ClassNotFoundException: sun.misc.Unsafe
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 15 more
```

This isn't a priority for me, but wanted you to know.

--- a/simpleclient/src/main/java/io/prometheus/client/Striped64.java
+++ b/simpleclient/src/main/java/io/prometheus/client/Striped64.java
@@ -9,6 +9,8 @@
 package io.prometheus.client;
 
 import java.util.Random;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
 
 /**
  * A package-local class holding common representation and mechanics
@@ -94,22 +96,10 @@ abstract class Striped64 extends Number {
         Cell(long x) { value = x; }
 
         final boolean cas(long cmp, long val) {
-            return UNSAFE.compareAndSwapLong(this, valueOffset, cmp, val);
+            return CAS_VALUE.compareAndSet(this, cmp, val);
         }
 
-        // Unsafe mechanics
-        private static final sun.misc.Unsafe UNSAFE;
-        private static final long valueOffset;
-        static {
-            try {
-                UNSAFE = getUnsafe();
-                Class<?> ak = Cell.class;
-                valueOffset = UNSAFE.objectFieldOffset
-                        (ak.getDeclaredField("value"));
-            } catch (Exception e) {
-                throw new Error(e);
-            }
-        }
+        private static final AtomicLongFieldUpdater<Cell> CAS_VALUE = AtomicLongFieldUpdater.newUpdater(Cell.class, "value");
 
     }
 
@@ -155,14 +145,14 @@ abstract class Striped64 extends Number {
      * CASes the base field.
      */
     final boolean casBase(long cmp, long val) {
-        return UNSAFE.compareAndSwapLong(this, baseOffset, cmp, val);
+        return CAS_BASE.compareAndSet(this, cmp, val);
     }
 
     /**
      * CASes the busy field from 0 to 1 to acquire lock.
      */
     final boolean casBusy() {
-        return UNSAFE.compareAndSwapInt(this, busyOffset, 0, 1);
+        return CAS_BUSY.compareAndSet(this, 0, 1);
     }
 
     /**
@@ -287,50 +277,7 @@ abstract class Striped64 extends Number {
         }
     }
 
-    // Unsafe mechanics
-    private static final sun.misc.Unsafe UNSAFE;
-    private static final long baseOffset;
-    private static final long busyOffset;
-    static {
-        try {
-            UNSAFE = getUnsafe();
-            Class<?> sk = Striped64.class;
-            baseOffset = UNSAFE.objectFieldOffset
-                    (sk.getDeclaredField("base"));
-            busyOffset = UNSAFE.objectFieldOffset
-                    (sk.getDeclaredField("busy"));
-        } catch (Exception e) {
-            throw new Error(e);
-        }
-    }
+    private static final AtomicLongFieldUpdater<Striped64> CAS_BASE = AtomicLongFieldUpdater.newUpdater(Striped64.class, "base");
+    private static final AtomicIntegerFieldUpdater<Striped64> CAS_BUSY = AtomicIntegerFieldUpdater.newUpdater(Striped64.class, "busy");
 
-    /**
-     * Returns a sun.misc.Unsafe.  Suitable for use in a 3rd party package.
-     * Replace with a simple call to Unsafe.getUnsafe when integrating
-     * into a jdk.
-     *
-     * @return a sun.misc.Unsafe
-     */
-    private static sun.misc.Unsafe getUnsafe() {
-        try {
-            return sun.misc.Unsafe.getUnsafe();
-        } catch (SecurityException tryReflectionInstead) {}
-        try {
-            return java.security.AccessController.doPrivileged
-                    (new java.security.PrivilegedExceptionAction<sun.misc.Unsafe>() {
-                        public sun.misc.Unsafe run() throws Exception {
-                            Class<sun.misc.Unsafe> k = sun.misc.Unsafe.class;
-                            for (java.lang.reflect.Field f : k.getDeclaredFields()) {
-                                f.setAccessible(true);
-                                Object x = f.get(null);
-                                if (k.isInstance(x))
-                                    return k.cast(x);
-                            }
-                            throw new NoSuchFieldError("the Unsafe");
-                        }});
-        } catch (java.security.PrivilegedActionException e) {
-            throw new RuntimeException("Could not initialize intrinsics",
-                    e.getCause());
-        }
-    }
 }

